Tests/XcodeGenKitTests/ProjectGeneratorTests.swift

```diff
@@ -1581,6 +1581,1767 @@ class ProjectGeneratorTests: XCTestCase {
             }
         }
     }
+        
+    func testGenerateXcodeProjectWithCustomDependencyDestinations() throws {
+        
+        describe("generateXcodeProject") {
+            
+            func generateProjectForApp(withDependencies: [Dependency], targets: [Target], packages: [String: SwiftPackage] = [:]) throws -> PBXProj {
+                
+                let app = Target(
+                    name: "App",
+                    type: .application,
+                    platform: .macOS,
+                    dependencies: withDependencies
+                )
+                
+                let project = Project(
+                    name: "test",
+                    targets: targets + [app],
+                    packages: packages
+                )
+
+                return try project.generatePbxProj()
+            }
+            
+            $0.context("with target dependencies") {
+                $0.context("application") {
+                    
+                    let appA = Target(
+                        name: "appA",
+                        type: .application,
+                        platform: .macOS
+                    )
+                    let appB = Target(
+                        name: "appB",
+                        type: .application,
+                        platform: .macOS
+                    )
+                    
+                    $0.it("does not embed them without copy phase spec") {
+                        
+                        // given
+                        let dependencies = [
+                            Dependency(type: .target, reference: appA.name, embed: true),
```

Comment:
There are several target types that original XcodeGen was not embedding even when embed was set to true. Or maybe it puts them into different sections than copyFiles section? I am not sure and I have not found any tests describing original embedding scheme for different types of dependencies